Sources/Testing/Attachments/Test.Attachment.swift

```diff
@@ -93,98 +73,134 @@ extension Test.Attachment {
   ///   - preferredName: The preferred name of the attachment when writing it
   ///     to a test report or to disk. If `nil`, the testing library attempts
   ///     to derive a reasonable filename for the attached value.
-  ///   - sourceLocation: The source location of the call to this initializer.
-  ///     This value is used when recording issues associated with the
-  ///     attachment.
-  public init(
-    _ attachableValue: some Test.Attachable & Sendable & Copyable,
-    named preferredName: String? = nil,
-    sourceLocation: SourceLocation = #_sourceLocation
-  ) {
-    let preferredName = preferredName ?? Self.defaultPreferredName
-    self.init(_attachableValue: attachableValue, preferredName: preferredName, sourceLocation: sourceLocation)
+  public init(_ attachableValue: consuming AttachableValue, named preferredName: String? = nil) {
+    self.attachableValue = attachableValue
+    self.preferredName = preferredName ?? Self.defaultPreferredName
   }
-#endif
+}
 
-  /// Attach this instance to the current test.
+@_spi(Experimental) @_spi(ForToolsIntegrationOnly)
+extension Test.Attachment where AttachableValue == any Test.Attachable & Sendable /* & Copyable rdar://137614425 */ {
+  /// Create a type-erased attachment from an instance of ``Test/Attachment``.
   ///
-  /// An attachment can only be attached once.
-  public consuming func attach() {
-    Event.post(.valueAttached(self))
+  /// - Parameters:
+  ///   - attachment: The attachment to type-erase.
+  fileprivate init(_ attachment: Test.Attachment<some Test.Attachable & Sendable & Copyable>) {
+    self.init(
+      attachableValue: attachment.attachableValue,
+      fileSystemPath: attachment.fileSystemPath,
+      preferredName: attachment.preferredName
+    )
   }
 }
+#endif
 
-// MARK: - Non-sendable and move-only attachments
-
-/// A type that stands in for an attachable type that is not also sendable.
-private struct _AttachableProxy: Test.Attachable, Sendable {
-  /// The result of `withUnsafeBufferPointer(for:_:)` from the original
-  /// attachable value.
-  var encodedValue = [UInt8]()
-
-  var estimatedAttachmentByteCount: Int?
+// MARK: - Attaching an attachment to a test (etc.)
 
-  func withUnsafeBufferPointer<R>(for attachment: borrowing Test.Attachment, _ body: (UnsafeRawBufferPointer) throws -> R) throws -> R {
-    try encodedValue.withUnsafeBufferPointer(for: attachment, body)
+extension Test.Attachment where AttachableValue: Test.Attachable & Sendable & Copyable {
+  /// Attach this instance to the current test.
+  ///
+  /// - Parameters:
+  ///   - sourceLocation: The source location of the call to this function.
+  ///
+  /// An attachment can only be attached once.
+  @_documentation(visibility: private)
+  public consuming func attach(sourceLocation: SourceLocation = #_sourceLocation) {
+    let attachmentCopy = Test.Attachment<any Test.Attachable & Sendable /* & Copyable rdar://137614425 */>(self)
```

Comment:
This posts an event but doesn't actually attach? Should it match the logic here: https://github.com/swiftlang/swift-testing/pull/814/files#diff-6feef57224f4a66aab636c27007b774e7a9bcf464b4383ba0976f88baca2fab3R134 ?

Reply:
This logic is correct. The difference from line 134 is that that one is just calling into this function, having gotten a copy of the attachment with a sendable/copyable value in it.